### DISCUSSION TITLE
Easy way to create a document with multiple authors

### DISCUSSION THREAD

**Discussion by awa5114** [papis/papis#934] [Q&A]
I was wondering whether it would be possible to add multiple authors to an author list when creating a new document all in one go and without using cumbersome dictionary syntax with family name and so forth? I notice that if I do the following:

```
papis add --set type "inproceedings" --set title "Urban Water Balance Analsis by Holistic Approach" --set author "Li, Shuping and Yu, Weiming"
```

The `info.yaml` ends up looking like this, and the authors are lumped together into one field:

```
author: Li, Shuping and Yu, Weiming
files: []
papis_id: 935e958bcba7805261e377bfff6d3e55
ref: Urban_Water_Bal_Li_Sh
time-added: 2024-11-04-10:08:45
title: Urban Water Balance Analsis by Holistic Approach
type: inproceedings
```

This won't do for a `ref-format` like so:

```
ref-format = {doc[author_list][0][family]}{doc[year]}
```

I'm aware I could add the author list as a python dictionary but that's cumbersome in my opinion:

```
papis update --set author_list "[{'family': 'Li', 'given': 'Shuping'}, {'family': 'Yu', 'given': 'Weiming'}]"
```

How do I get it so that the authors are separated with a relatively straightforward command?

**Comment by jghauser** [comment]
I don't think there's an easy/straightforward way to do this. In fact, I've been sufficiently annoyed with manually adding docs to my library that I created [a little script to help me](https://gist.github.com/jghauser/9e1a98b52f63a9b2c9c82b5668884b74). It basically prompts for various fields depending on the type of document (otherwise I always forget some), providing a sort of 'manual adding wizard'. Maybe you can find some inspiration there :). Though I have to admit, something more built-in would be great!

**Reply by jghauser** [reply]
Oh, there's probably an easier way to do this with `papis doctor` which has a function to create `author_list` from `author` fields.

EDIT: A little more info: I think if you set the [`auto-doctor`](https://papis.readthedocs.io/en/latest/configuration.html#confval-auto-doctor) setting to `true`, it should automatically create `author_list` from `author` when adding documents. I haven't tested it though!

**Comment by alexfikl** [comment]
Just to add another way that I usually do this: using the `--edit` flag
```
papis add --edit --set ...
```
then after everything it set up, it brings up the editor to add additional fields. Not very nice, but it happens rarely enough that I never got annoyed.
